### DISCUSSION TITLE
Document using `mutate` with options without clearing the cache

### DISCUSSION THREAD

**Discussion by colinking** [vercel/swr#2490] [Q&A]
# Bug report

## Description / Observed Behavior

Calling `mutate(key, undefined, {...})` will clear the cache for `key` while `mutate(key)` will not.

While writing this issue, I found a workaround. You can pass a `MutatorCallback` as `mutate(key, (data) => data, {...})` and that retains the existing cache value.

## Expected Behavior

I would expect the default behavior of `mutate` to preserve the existing stale cache data (for "stale-while-revalidate" semantics), but the `mutate` docs only show examples where `undefined` is passed. Documentation would help.

Ideally, `mutate(key, undefined)` is the same as `mutate(key, (data) => data)` to provide SWR behavior out-of-the-box when passing options, but that may be considered a breaking change.

## Repro Steps / Code Example

Here's an example [CodeSandbox](https://codesandbox.io/p/sandbox/pensive-sunset-4feb6x?selection=%5B%7B%22endColumn%22%3A27%2C%22endLineNumber%22%3A7%2C%22startColumn%22%3A27%2C%22startLineNumber%22%3A7%7D%5D&file=%2Fpackage.json).

Notice that clicking on the `Mutate` button causes the data to flash briefly as `undefined`.

## Additional Context

SWR version: 2.0.3

**Comment by leocorder** [comment]
When calling `mutate(key)` you're telling SWR to set the cache value as expired for given key. The cache value is untouched until refetch is made.
When calling `mutate(key, undefined)` you're telling SWR to immediately update cache value to undefined, before refetch is made.
When calling `mutate(key, (data) => data)` you're telling SWR to immediately update cache value to `data`, before refetch is made. Note that `data` is not undefined, it's the currently cache value stored for that key. So your key will hold the current cache value until refetch is made. In practice, `mutate(key, (data) => data)` is the same as using `mutate(key)`.
